modules/internal/BytesStringCommon.chpl

```diff
@@ -839,4 +848,13 @@ module BytesStringCommon {
     }
     return hash:uint;
   }
+
+  inline proc setCodepoints(ref lhs: ?t1, rhs: ?t2) {
```

Comment:
I expect types to be `(ref lhs: string, rhs: string)`. This function is not applicable to anything else but strings and as such, we want the compiler to be able to catch any error where this was used otherwise.

Just as a side note, as it is now this would allow any two arguments of any type (not necessarily the same). If you want them to be of same type you should do `ref lhs: ?t1, rhs: t1`

Reply:
I would also make two other changes:
- Function name can be more verbose: `incrementCodepoints` maybe? Or you can make it `addCachedNumCodepoints` and make it return an `int` instead of changing `lhs`.
- We can make this function `private`, right? In other words, I don't expect it to be used outside of this module, but if you are using it in `String.chpl`, then leave it be. (And maybe add a comment to the private follow up issue, so that we can see if we can make it `private` with some more effort)